tests/test_other.py

```diff
@@ -7982,6 +7982,65 @@ def test_closure_cmdline_utf8_chars(self):
     create_file(externs, '')
     self.run_process([EMCC, test, '--closure=1', '--closure-args', '--externs "' + externs + '"'])
 
+  def test_closure_type_annotations(self):
+    # Verify that certain type annotations exist to allow closure to avoid
+    # ambiguity and maximize optimization opportunities in user code.
+    # Currently we check for a fixed set of know attribute names which
+    # have been reported as unannoted in the past:
+
+    attributes = ['put', 'getContext', 'contains', 'stopPropagation', 'pause']
+    methods = ''
+    for attribute in attributes:
+      methods += '''
+        this.%s = function() {
+          console.error("my %s");
+        };
+      ''' % (attribute, attribute)
```

Comment:
f-string?

Reply:
f-strings are not always great for strings that contain literal `{` because they need to be escaped.  I think case I think it maybe make sense though.